contracts/Permissions.sol

```diff
@@ -9,6 +9,13 @@ enum Clearance {
     FUNCTION
 }
 
+enum ExecutionMode {
+    BARE,
+    SEND,
+    DELEGATE,
+    BOTH
```

Comment:
`BOTH` should be `ALL`, no? Assuming that it implies that you have all of the above permissions.
I guess you could argue that `SEND` with 0 value is equivalent to `BARE`.

Does `DELEGATE` also imply `BARE`? i.e. if you are allowed to delegate call are you always allowed to call?

Reply:
Not sure I get your question. So this refers to the "extras" you can do while executing, "execution modes"

It has no impact on actually allowing/disallowing calls. It's transversal:

BARE ->             send 🚫  delegate 🚫 
SEND ->            send ✅  delegate 🚫 
DELEGATE ->    send 🚫  delegate ✅ 
BOTH->             send ✅  delegate ✅

Reply:
(not loving the names on the enum)

Reply:
> Not sure I get your question.

Put another way. Is it possible to allow delegate calls to a function without allowing calls to a function?

It seems like the options are:
CALL -> CALL :white_check_mark: send :no_entry_sign: delegate call :no_entry_sign: 
SEND -> CALL :white_check_mark: send :white_check_mark:  delegate call :no_entry_sign: 
DELEGATE CALL -> CALL :white_check_mark: send :no_entry_sign: delegate call :white_check_mark: 
ALL -> CALL :white_check_mark: send :white_check_mark: delegate call :white_check_mark:

Reply:
Ah, got you now. Yes I gave some thought and attention to this detail.

The answer is no you can not! And it behaves consistently for both two levels: for TARGET and FUNCTION.

Are these assumptions correct?

Reply:
So the clearance to call a target with delegateCall comes from:
* [targetAddress].clearance == TARGET   AND
* [targetAddress].mode must allow DelegateCall (i.e., DELEGATE or BOTH)


The clearance to call a function with delegateCall comes from:
* targets[targetAddress].clearance == FUNCTION AND
* functions[targetAddress+functionSig] (the scopeConfig) must allow the current call AND
*  functions[targetAddress+functionSig] (the scopeConfig) must contain a mode that allows DelegateCall (DELEGATE or BOTH)

Depends on the granularity the user chose to configure: he could have AllowedTarget, or AllowedTargetPartially

Reply:
This is what I mean when I say mode (previously canSend and canDelegate) are transversal checks.

They are not an explicit allowance.

They simply allow that OperationMode, in case the call is explicitly authorized.

Hope this makes sense

Reply:
How about this?
```solidity
enum ExecutionOptions {
    NONE,
    SEND,
    DELEGATECALL,
    BOTH
}
```

Reply:
Will adjust soon, thanks for the suggestion